lib/npg/model/instrument_format.pm

```diff
@@ -118,16 +137,30 @@ sub _obtain_numerical_name_part {
 sub current_instruments_by_format {
   my ( $self ) = @_;
 
+  my $cgi = $self->util()->cgi();
+  my $filter_lab = $cgi->param('filter_lab');
+
   if ( ! $self->{current_instruments_by_format} ) {
     my $href = {};
     foreach my $format ( @{ $self->current_instrument_formats() } ) {
       my $model = $format->model();
       $model = $model eq q{HK} ? q{GA-II} : $model;
-      my @ordered = map  { $_->[0] }
-                    sort { $a->[1] <=> $b->[1] }
-                    map  { [ $_, $self->_obtain_numerical_name_part( $_->name() ) ] } @{ $format->current_instruments() };
-      if ( scalar @ordered ) {
-        $href->{$model} = \@ordered;
+      if ($filter_lab) {
```

Comment:
I advise against duplicating complex functional statements when only the input list changes.
```suggestion
      my @instruments;
      if ($filter_lab) {
        @instruments = $format->current_instruments_from_lab($filter_lab);
      } else {
        @instruments = $format->current_instruments();
      }
      ... map sort map...
```